lib/editor/actions/map/stopStrategies.js

```diff
@@ -482,26 +507,29 @@ function extendPatternToPoint (pattern, endPoint, newEndPoint, stop = null, spli
   }
 }
 
-export function removeStopFromPattern (pattern: Pattern, stop: GtfsStop, index: number) {
+export function removeStopFromPattern (pattern: Pattern, stop: GtfsLocation | GtfsStop, index: number) {
   return async function (dispatch: dispatchFn, getState: getStateFn) {
     const {controlPoints, patternSegments} = getControlPoints(getState())
     const clonedControlPoints = clone(controlPoints)
     const clonedPatternSegments = clone(patternSegments)
     const {shapePoints} = pattern
     const patternStops = [...pattern.patternStops]
+    const patternLocations = [...pattern.patternLocations]
+    const patternHalts = mergePatternHalts(patternStops, patternLocations)
+    const stopIsStop = stop.hasOwnProperty('stopId')
```

Comment:
Is there an assumption that stopId can be null for regular stops?

Reply:
We do this because a stop_id of 0 equates to false!